flink-runtime/src/test/java/org/apache/flink/runtime/io/network/NettyShuffleEnvironmentCleanupTest.java

```diff
@@ -0,0 +1,181 @@
+import org.apache.flink.runtime.entrypoint.StandaloneSessionClusterEntrypoint;
+import org.apache.flink.runtime.taskexecutor.TaskManagerRunner;
+import org.apache.flink.runtime.testutils.TestJvmProcess;
+import org.apache.flink.runtime.zookeeper.ZooKeeperResource;
+import org.apache.flink.util.OperatingSystem;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.time.Duration;
+import java.util.Map;
+
+import static org.apache.flink.runtime.testutils.CommonTestUtils.createTemporaryLog4JProperties;
+import static org.apache.flink.runtime.testutils.CommonTestUtils.getJavaCommandPath;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.junit.Assume.assumeTrue;
+
+/**
+ * Verifies whether netty shuffle releases all the resources on shutdown, like the temporary directories.
+ */
+public class NettyShuffleEnvironmentCleanupTest {
+	private static final Duration TEST_TIMEOUT = Duration.ofSeconds(10);
+
+	@Rule
+	public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+	@Rule
+	public final ZooKeeperResource zooKeeperResource = new ZooKeeperResource();
+
+	@Test
+	public void testRemovingTmpDirectoriesOnSignals() throws Exception {
+		assumeTrue(OperatingSystem.isLinux()
+				|| OperatingSystem.isFreeBSD()
+				|| OperatingSystem.isSolaris()
+				|| OperatingSystem.isMac());
+
+		File confDir = temporaryFolder.newFolder();
+		File confFile = new File(confDir + "/flink-conf.yaml");
+
+		File taskManagerTmpDir = temporaryFolder.newFolder();
+
+		Configuration config = new Configuration();
+		config.setString(JobManagerOptions.ADDRESS, "localhost");
+		config.setString(RestOptions.BIND_PORT, "0");
+		config.setString(HighAvailabilityOptions.HA_MODE, "zookeeper");
+		config.setString(HighAvailabilityOptions.HA_ZOOKEEPER_QUORUM, zooKeeperResource.getConnectString());
+		config.setString(HighAvailabilityOptions.HA_STORAGE_PATH, temporaryFolder.newFolder().getAbsolutePath());
+		config.setString(CoreOptions.TMP_DIRS, taskManagerTmpDir.getAbsolutePath());
+		config.setString(TaskManagerOptions.MANAGED_MEMORY_SIZE, "4m");
+		config.setInteger(NettyShuffleEnvironmentOptions.NETWORK_NUM_BUFFERS, 100);
+		config.setString(TaskManagerOptions.TOTAL_FLINK_MEMORY, "512m");
+
+		try (FileOutputStream fos = new FileOutputStream(confFile);
+			BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(fos))) {
+			for (Map.Entry<String, String> e : config.toMap().entrySet()) {
+				writer.write(e.getKey());
+				writer.write(": ");
+				writer.write(e.getValue());
+				writer.newLine();
+			}
+
+			writer.flush();
+		}
+
+		TaskManagerProcess taskManagerProcess = null;
+
+		try (final StandaloneSessionClusterEntrypoint clusterEntrypoint = new StandaloneSessionClusterEntrypoint(config)) {
```

Comment:
It is too heavy-height to start the cluster besides the preparation of above configuration and writer. We can construct the `FileChannelManagerImpl` via `TestJvmProcess` directly for simple. Actually we only want to verify that the new added shutdown hook to `FileChannelManagerImpl` work correctly, no matter which component covers it.

Reply:
In addition, we should rename `NettyShuffleEnvironmentCleanupTest` to `FileChannelManagerImplTest` instead.

Reply:
Agree with that and I have modified the test accordingly.